spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/core/SpannerObjectMapperImplTests.java

```diff
@@ -0,0 +1,226 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core;
+
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.Date;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.Mutation.WriteBuilder;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Struct;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.spanner.ValueBinder;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerColumn;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerMappingContext;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerObjectMapper;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerObjectMapperImpl;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerTable;
+import org.springframework.data.annotation.Id;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyDouble;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Chengyuan Zhao
+ */
+@RunWith(SpringRunner.class)
+public class SpannerObjectMapperImplTests {
+
+	private SpannerObjectMapper objectMapper;
+
+	@Before
+	public void setUp() {
+		this.objectMapper = new SpannerObjectMapperImpl(new SpannerMappingContext());
+	}
+
+	@Test
+	public void writeTest() throws ClassNotFoundException {
```

Comment:
This test is ensuring that the correctly typed "to" methods are called. This is vital, because that's where our responsibility leaves and it's up to the Spanner client lib. The key point of the write-mapping is to make sure the correctly typed functions are mapped.  I will rename the fields.